src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -11176,6 +11176,45 @@ public void InvalidReturnInCompoundAssignment()
             );
         }
 
+        [Fact, WorkItem(49639, "https://github.com/dotnet/roslyn/issues/49639")]
+        public void CompareToNullWithNestedUnconstrainedTypeParameter()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+using System;
+using System.Threading.Tasks;
+unsafe
+{
+    test<int>(null);
+
+    static void test<T>(delegate*<T, void> f)
```

Comment:
>static void test<T>(delegate*<T, void> f) [](start = 4, length = 41)

Perhaps should cover non-generic case as well for the purpose of catching possible regression in the future.

Reply:
See `FunctionPointerTypePatternIsNull`: [https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Test/Semantic/Semantics/FunctionPointerTests.cs#L2340](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Test/Semantic/Semantics/FunctionPointerTests.cs#L2340).

---
In reply to: [541365340](https://github.com/dotnet/roslyn/pull/49915#discussion_r541365340) [](ancestors = 541365340)